WireDomain/Sources/WireDomain/Event Processing/ConversationEventProcessor/ConversationMemberUpdateEventProcessor.swift

```diff
@@ -32,9 +33,48 @@ protocol ConversationMemberUpdateEventProcessorProtocol {
 
 struct ConversationMemberUpdateEventProcessor: ConversationMemberUpdateEventProcessorProtocol {
 
-    func processEvent(_: ConversationMemberUpdateEvent) async throws {
-        // TODO: [WPB-10170]
-        assertionFailure("not implemented yet")
+    let conversationRepository: any ConversationRepositoryProtocol
+    let userRepository: any UserRepositoryProtocol
+    let localStore: any ConversationLocalStoreProtocol
+
+    func processEvent(_ event: ConversationMemberUpdateEvent) async throws {
+        let senderID = event.senderID
+        let conversationID = event.conversationID
+        let memberChange = event.memberChange
+        let muteStatus = memberChange.newMuteStatus
+        let muteStatusDate = memberChange.muteStatusReferenceDate
+        let archivedStatus = memberChange.newArchivedStatus
+        let archivedStatusDate = memberChange.archivedStatusReferenceDate
+
+        let conversation = await conversationRepository.fetchOrCreateConversation(
+            with: conversationID.uuid,
+            domain: conversationID.domain
+        )
+
+        let isSelfUser = try await userRepository.isSelfUser(
+            id: senderID.uuid,
+            domain: senderID.domain
+        )
+
+        if isSelfUser {
+            await localStore.updateMemberStatus(
+                mutedStatusInfo: (muteStatus, muteStatusDate),
+                archivedStatusInfo: (archivedStatus, archivedStatusDate),
+                for: conversation
+            )
+        }
+
+        guard let role = event.memberChange.newRoleName else {
+            return
+        }
+
+        await conversationRepository.addParticipantToConversation(
+            conversationID: conversationID.uuid,
+            conversationDomain: conversationID.domain,
+            participantID: senderID.uuid,
+            participantDomain: senderID.domain,
+            participantRole: role
+        )
```

Comment:
So the update for other users is only their role? This method suggests that they are new members (I would expect then the participant add event) that need adding, but I'm feel like we should be simply updating their existing role with a new one.

Reply:
It updates its role if the user is already part of the conversation or adds the user to the conversation if not. The legacy method `func addParticipantAndUpdateConversationState(user: ZMUser, role: Role? = nil)` automatically handles both logic. This is a bit confusing, I agree, however in legacy codebase this is how this event is currently processed.

I rename some methods and added comments in https://github.com/wireapp/wire-ios/pull/2091/commits/53edd4d6253f4c3bac2a4973f06d518d1a476786 to make it more clear.